### DISCUSSION TITLE
Client Data Normalization and Caching

### DISCUSSION THREAD

**Discussion by danielyogel** [trpc/trpc#707] [Q&A]
While I find tRPC's client amazing for what is does, I miss some functionality that larger clients (say Apollo client) provides - mostly data normalization & caching. 

I wonder if any of normalization / caching are part of tRPC's vision, and if not - what other solutions do you combine with tRPC?

**Comment by KATT** [comment]
## Caching

I have not created any caching link or anything like that as react query handles that for me automatically. If you're using React, use `@trpc/react`.

You can do a cache link yourself pretty easily [using links](https://trpc.io/docs/links), I'll make an example and link here

## Normalization

Can you elaborate on this?

**Reply by KATT** [reply]
For caching have a look at https://github.com/trpc/trpc/pull/708 and read the docs on links https://trpc.io/docs/links

**Reply by KATT** [reply]
For normalization, I've made a proof of concept back in https://github.com/trpc/trpc/pull/58
